src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp

```diff
@@ -105,35 +105,34 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
 
   GEOS_HOST_DEVICE
   void computePorosity( real64 const & deltaPressureFromBeginningOfTimeStep,
-                        real64 const & deltaPressureFromLastIteration,
                         real64 const & deltaTemperatureFromBeginningOfTimeStep,
-                        real64 const & deltaTemperatureFromLastIteration,
                         real64 const & porosity_n,
                         real64 const & referencePorosity,
                         real64 & porosity,
                         real64 & dPorosity_dPressure,
                         real64 & dPorosity_dTemperature,
                         real64 const & biotCoefficient,
                         real64 const & thermalExpansionCoefficient,
-                        real64 const & meanEffectiveStressIncrement_k,
+                        real64 const & averageMeanTotalStressIncrement_k,
                         real64 const & bulkModulus ) const
   {
     real64 const biotSkeletonModulusInverse = (biotCoefficient - referencePorosity) / m_grainBulkModulus;
     real64 const porosityThermalExpansion = 3 * thermalExpansionCoefficient * ( biotCoefficient - referencePorosity );
     real64 const fixedStressPressureCoefficient = biotCoefficient * biotCoefficient / bulkModulus;
     real64 const fixedStressTemperatureCoefficient = 3 * biotCoefficient * thermalExpansionCoefficient;
 
+    // total stress formulation for porosity update
     porosity = porosity_n
-               + biotCoefficient * meanEffectiveStressIncrement_k / bulkModulus // change due to stress increment (at the previous
-                                                                                // sequential iteration)
+               + biotCoefficient * averageMeanTotalStressIncrement_k / bulkModulus // change due to stress increment (at the previous
+                                                                                   // sequential iteration)
                + biotSkeletonModulusInverse * deltaPressureFromBeginningOfTimeStep // change due to pressure increment
                - porosityThermalExpansion * deltaTemperatureFromBeginningOfTimeStep; // change due to temperature increment
     dPorosity_dPressure = biotSkeletonModulusInverse;
     dPorosity_dTemperature = -porosityThermalExpansion;
 
     // Fixed-stress part
-    porosity += fixedStressPressureCoefficient * deltaPressureFromLastIteration   // fixed-stress pressure term
-                + fixedStressTemperatureCoefficient * deltaTemperatureFromLastIteration;   // fixed-stress temperature term
+    porosity += fixedStressPressureCoefficient * deltaPressureFromBeginningOfTimeStep // fixed-stress pressure term
+                + fixedStressTemperatureCoefficient * deltaTemperatureFromBeginningOfTimeStep; // fixed-stress temperature term
     dPorosity_dPressure += fixedStressPressureCoefficient;
     dPorosity_dTemperature += fixedStressTemperatureCoefficient;
```

Comment:
Every time I go through these lines I got lost and I always have the impression that a pressure (and temperature) term is missing in the lines 124-129. Could you please post here a quick derivation? It would be very helpful.

Reply:
![image](https://github.com/GEOS-DEV/GEOS/assets/2421323/9cb08e26-b56b-456e-8168-17514379124b)
and temperature is similar